tests/Unit/Framework/MockRuntimeSystem.hpp

```diff
@@ -176,14 +184,19 @@ class MockRuntimeSystem {
     // passed the mock node and mock core associated with that cache.
     // The actual chares held in each cache are emplaced by the user explicitly
     // on a chosen node and core, via `emplace_component`.
-    for (const auto& [node,local_core] : mock_nodes_and_local_cores_) {
-      const size_t global_core = mock_global_cores_.at(node).at(local_core);
+    for (const auto& [global_core_id, node_and_local_core] :
+         mock_nodes_and_local_cores_) {
+      const size_t global_core = global_core_id.value;
       mutable_caches_.at(global_core) =
-                std::make_unique<Parallel::MutableGlobalCache<Metavariables>>(
-                    serialize_and_deserialize(mutable_cache_contents));
+          std::make_unique<Parallel::MutableGlobalCache<Metavariables>>(
+              serialize_and_deserialize(mutable_cache_contents));
       caches_.at(global_core) = std::make_unique<GlobalCache>(
           serialize_and_deserialize(cache_contents),
           mutable_caches_.at(global_core).get());
+      // Note that lambdas cannot capture structured bindings,
+      // so we define node and local_core here.
+      const auto& node = node_and_local_core.first.value;
+      const auto& local_core = node_and_local_core.second.value;
```

Comment:
Ah this explains my previous comment :) Probably better to fix the earlier commit so it compiles.